framework/src/base/MooseVariableBase.C

```diff
@@ -46,9 +46,24 @@ MooseVariableBase::name() const
   return _sys.system().variable(_var_num).name();
 }
 
+std::vector<dof_id_type>
+MooseVariableBase::allDofIndices() const
```

Comment:
This should be done using the existing threaded object and a parallel reduction.